// @aemforms/af-core/lib/esm/utils/FormCreationUtils.js
/*************************************************************************
* ADOBE CONFIDENTIAL
* ___________________
*
* Copyright 2022 Adobe
* All Rights Reserved.
*
* NOTICE: All information contained herein is, and remains
* the property of Adobe and its suppliers, if any. The intellectual
* and technical concepts contained herein are proprietary to Adobe
* and its suppliers and are protected by all applicable intellectual
* property laws, including trade secret and copyright laws.
* Dissemination of this information or reproduction of this material
* is strictly forbidden unless prior written permission is obtained
* from Adobe.

* Adobe permits you to use and modify this file solely in accordance with
* the terms of the Adobe license agreement accompanying it.
*************************************************************************/

import { InstanceManager } from '../InstanceManager.js';
import { Fieldset } from '../Fieldset.js';
import { isRepeatable, isFile, isCheckbox, isCheckboxGroup, isDateField } from './JsonUtils.js';
import FileUpload from '../FileUpload.js';
import Checkbox from '../Checkbox.js';
import CheckboxGroup from '../CheckboxGroup.js';
import DateField from '../DateField.js';
import Field from '../Field.js';
import '../BaseNode-d78cc1b0.js';
import '../controller/Events.js';
import './DataRefParser.js';
import '../data/DataGroup.js';
import '../data/DataValue.js';
import '../data/EmptyDataValue.js';
import '../Container.js';
import '../Scriptable.js';
import '../types/Json.js';
import './SchemaUtils.js';
import './FormUtils.js';
import '../FileObject.js';
import './ValidationUtils.js';
import '../types/Model.js';
import '@aemforms/af-formatters';

const alternateFieldTypeMapping = {
    'text': 'text-input',
    'number': 'number-input',
    'email': 'text-input',
    'file': 'file-input',
    'range': 'range',
    'textarea': 'multiline-input'
};
class FormFieldFactoryImpl {
    createField(child, _options) {
        let retVal;
        const options = {
            ..._options,
            fieldFactory: this
        };
        child.fieldType = child.fieldType ? (child.fieldType in alternateFieldTypeMapping ?
            alternateFieldTypeMapping[child.fieldType] : child.fieldType)
            : 'text-input';
        if (isRepeatable(child)) {
            const newChild = {
                ...child,
                ...('items' in child && { 'type': 'object' }),
                minOccur: undefined,
                maxOccur: undefined,
                repeatable: undefined,
                name: undefined
            };
            const newJson = {
                ...{
                    minItems: child.minOccur || 0,
                    maxItems: child.maxOccur || -1,
                    fieldType: child.fieldType,
                    type: 'array',
                    name: child.name,
                    dataRef: child.dataRef
                },
                ...{
                    'items': [newChild]
                }
            };
            retVal = new InstanceManager(newJson, options);
        }
        else if ('items' in child) {
            retVal = new Fieldset(child, options);
        }
        else {
            if (isFile(child) || child.fieldType === 'file-input') {
                retVal = new FileUpload(child, options);
            }
            else if (isCheckbox(child)) {
                retVal = new Checkbox(child, options);
            }
            else if (isCheckboxGroup(child)) {
                retVal = new CheckboxGroup(child, options);
            }
            else if (isDateField(child)) {
                retVal = new DateField(child, options);
            }
            else {
                retVal = new Field(child, options);
            }
        }
        return retVal;
    }
}
const FormFieldFactory = new FormFieldFactoryImpl();

export { FormFieldFactory };
