// @react-spectrum/link/dist/main.js
require("./main.css");
var $fo5qX$reactspectrumutils = require("@react-spectrum/utils");
var $fo5qX$reactariafocus = require("@react-aria/focus");
var $fo5qX$reactariautils = require("@react-aria/utils");
var $fo5qX$react = require("react");
var $fo5qX$reactariainteractions = require("@react-aria/interactions");
var $fo5qX$reactarialink = require("@react-aria/link");
var $fo5qX$reactspectrumprovider = require("@react-spectrum/provider");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Link", () => $1148fb7fc40f1ce2$export$a6c7ac8248d6e38a);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



var $094ce2289235a9fa$exports = {};

$parcel$export($094ce2289235a9fa$exports, "spectrum-Link", () => $094ce2289235a9fa$export$d85dd8b94975f388, (v) => $094ce2289235a9fa$export$d85dd8b94975f388 = v);
$parcel$export($094ce2289235a9fa$exports, "is-hovered", () => $094ce2289235a9fa$export$b8813cd5d7824ce7, (v) => $094ce2289235a9fa$export$b8813cd5d7824ce7 = v);
$parcel$export($094ce2289235a9fa$exports, "focus-ring", () => $094ce2289235a9fa$export$f39a09f249340e2a, (v) => $094ce2289235a9fa$export$f39a09f249340e2a = v);
$parcel$export($094ce2289235a9fa$exports, "is-disabled", () => $094ce2289235a9fa$export$d35bc1e505d1ebbf, (v) => $094ce2289235a9fa$export$d35bc1e505d1ebbf = v);
$parcel$export($094ce2289235a9fa$exports, "spectrum-Link--secondary", () => $094ce2289235a9fa$export$ab09ed292dcb9a79, (v) => $094ce2289235a9fa$export$ab09ed292dcb9a79 = v);
$parcel$export($094ce2289235a9fa$exports, "spectrum-Link--overBackground", () => $094ce2289235a9fa$export$679b6267148797da, (v) => $094ce2289235a9fa$export$679b6267148797da = v);
$parcel$export($094ce2289235a9fa$exports, "spectrum-Link--quiet", () => $094ce2289235a9fa$export$81921ea93435e997, (v) => $094ce2289235a9fa$export$81921ea93435e997 = v);
var $094ce2289235a9fa$export$d85dd8b94975f388;
var $094ce2289235a9fa$export$b8813cd5d7824ce7;
var $094ce2289235a9fa$export$f39a09f249340e2a;
var $094ce2289235a9fa$export$d35bc1e505d1ebbf;
var $094ce2289235a9fa$export$ab09ed292dcb9a79;
var $094ce2289235a9fa$export$679b6267148797da;
var $094ce2289235a9fa$export$81921ea93435e997;
$094ce2289235a9fa$export$d85dd8b94975f388 = "spectrum-Link_5a03c7";
$094ce2289235a9fa$export$b8813cd5d7824ce7 = "is-hovered_5a03c7";
$094ce2289235a9fa$export$f39a09f249340e2a = "focus-ring_5a03c7";
$094ce2289235a9fa$export$d35bc1e505d1ebbf = "is-disabled_5a03c7";
$094ce2289235a9fa$export$ab09ed292dcb9a79 = "spectrum-Link--secondary_5a03c7";
$094ce2289235a9fa$export$679b6267148797da = "spectrum-Link--overBackground_5a03c7";
$094ce2289235a9fa$export$81921ea93435e997 = "spectrum-Link--quiet_5a03c7";





function $1148fb7fc40f1ce2$export$a6c7ac8248d6e38a(props) {
    props = (0, $fo5qX$reactspectrumprovider.useProviderProps)(props);
    props = (0, $fo5qX$reactspectrumutils.useSlotProps)(props, "link");
    let { variant: variant = "primary" , isQuiet: isQuiet , children: children , href: // @ts-ignore
    href  } = props;
    let { styleProps: styleProps  } = (0, $fo5qX$reactspectrumutils.useStyleProps)(props);
    let { hoverProps: hoverProps , isHovered: isHovered  } = (0, $fo5qX$reactariainteractions.useHover)({});
    if (href) console.warn("href is deprecated, please use an anchor element as children");
    let ref = (0, $fo5qX$react.useRef)();
    let { linkProps: linkProps  } = (0, $fo5qX$reactarialink.useLink)({
        ...props,
        elementType: typeof children === "string" ? "span" : "a"
    }, ref);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($fo5qX$react))).createElement((0, $fo5qX$reactariafocus.FocusRing), {
        focusRingClass: (0, $fo5qX$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($094ce2289235a9fa$exports))), "focus-ring")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($fo5qX$react))).cloneElement((0, $fo5qX$reactspectrumutils.getWrappedElement)(children), {
        ...styleProps,
        ...(0, $fo5qX$reactariautils.mergeProps)(linkProps, hoverProps),
        ref: ref,
        className: (0, $fo5qX$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($094ce2289235a9fa$exports))), "spectrum-Link", {
            "spectrum-Link--quiet": isQuiet,
            [`spectrum-Link--${variant}`]: variant,
            "is-hovered": isHovered
        }, styleProps.className)
    }));
}
