// @adobe/exc-app/capabilityapi.js
"use strict";
/*************************************************************************
 * Copyright 2021 Adobe
 * All Rights Reserved.
 *
 * NOTICE: Adobe permits you to use, modify, and distribute this file in
 * accordance with the terms of the Adobe license agreement accompanying
 * it. If you have received this file from a source other than Adobe,
 * then your use, modification, or distribution of it requires the prior
 * written permission of Adobe.
 **************************************************************************/
Object.defineProperty(exports, "__esModule", { value: true });
exports.CapabilityIds = exports.CapabilityNames = void 0;
/**
 * API to request capability icons by id (or secondarily by display name).
 *
 * ***Import:***
 *
 * ```typescript
 * import capabilityApi from '@adobe/exc-app/capabilityapi';
 * ```
 *
 * ***Default export:***
 *
 * [CapabilityApi](../interfaces/capability.capabilityapi.md#interface-capabilityapi)
 *
 * ***Usage:***
 * These APIs return promises, not properties, and will need to be
 * asynchronously loaded by calling `get`.
 *
 * ```typescript
 * const [home, assets, offer] = await Promise.all([
 *   capabilityApi.get('home'),
 *   capabilityApi.get('assets'),
 *   capabilityApi.get('offer')
 * ]);
 *
 * ```
 *
 * API for fetching capability icon data.
 * @packageDocumentation
 * @module capabilityapi
 */
const Global_1 = require("./src/Global");
// This list is dynamically generated in preBuild
var CapabilityNames;
(function (CapabilityNames) {
    CapabilityNames["ACCOUNTPROFILE"] = "Profiles";
    CapabilityNames["ACCOUNTSEGMENTS"] = "Segments";
    CapabilityNames["ADMIN"] = "Configurations";
    CapabilityNames["ALERT"] = "Alerts";
    CapabilityNames["APPCONFIGURATIONS"] = "App Configurations";
    CapabilityNames["APPLICATIONS"] = "Applications";
    CapabilityNames["APPSURFACES"] = "App Surfaces";
    CapabilityNames["ASSETS"] = "Assets";
    CapabilityNames["ASSURANCE"] = "Assurance";
    CapabilityNames["CAMPAIGNS"] = "Campaigns";
    CapabilityNames["CLIENTPROPERTIES"] = "Tags";
    CapabilityNames["COMPONENTS"] = "Components";
    CapabilityNames["CONTENTTEMPLATES"] = "Content Templates";
    CapabilityNames["CONTROLRULES"] = "Rules";
    CapabilityNames["DASHBOARD"] = "Dashboards";
    CapabilityNames["DATACLEANSING"] = "Data Hygiene";
    CapabilityNames["DATACOLLECTIONMONITORING"] = "Monitoring";
    CapabilityNames["DATAFLOW"] = "Dataflows";
    CapabilityNames["DATASET"] = "Datasets";
    CapabilityNames["DATASTREAMS"] = "Datastreams";
    CapabilityNames["DATASTREAMSBETA"] = "Datastreams (Beta)";
    CapabilityNames["DESTINATION"] = "Destinations";
    CapabilityNames["FRAGMENTS"] = "Fragments";
    CapabilityNames["HOME"] = "Home";
    CapabilityNames["IDENTITIES"] = "Identities";
    CapabilityNames["IDENTITY"] = "Identities";
    CapabilityNames["INITIATIVES"] = "Initiatives";
    CapabilityNames["JOURNEYS"] = "Journeys";
    CapabilityNames["LANDINGPAGES"] = "Landing pages";
    CapabilityNames["LICENSEUSAGE"] = "License Usage";
    CapabilityNames["MESSAGES"] = "Messages";
    CapabilityNames["MODEL"] = "Models";
    CapabilityNames["MONITORING"] = "Monitoring";
    CapabilityNames["NOTEBOOK"] = "Notebooks";
    CapabilityNames["OFFER"] = "Offers";
    CapabilityNames["OFFERACTIVITY"] = "Components";
    CapabilityNames["PERMISSIONSUI"] = "Permissions";
    CapabilityNames["PLACES"] = "Places";
    CapabilityNames["PLAYBOOKS"] = "Playbooks";
    CapabilityNames["POLICY"] = "Policies";
    CapabilityNames["PRIVACYAUDIT"] = "Audits";
    CapabilityNames["PRIVACYCONSOLE"] = "Privacy Console";
    CapabilityNames["PROFILE"] = "Profiles";
    CapabilityNames["QUERY"] = "Queries";
    CapabilityNames["REQUEST"] = "Requests";
    CapabilityNames["SANDBOX"] = "Sandboxes";
    CapabilityNames["SCHEMA"] = "Schemas";
    CapabilityNames["SEGMENT"] = "Segments";
    CapabilityNames["SERVERPROPERTIES"] = "Event Forwarding";
    CapabilityNames["SERVICE"] = "Services";
    CapabilityNames["SETTINGS"] = "Channels";
    CapabilityNames["SOURCE"] = "Sources";
    CapabilityNames["SUBSCRIPTIONS"] = "Subscription list";
    CapabilityNames["UNIFIEDTAGS"] = "UnifiedTags";
    CapabilityNames["WORKFLOW"] = "Workflows";
})(CapabilityNames = exports.CapabilityNames || (exports.CapabilityNames = {}));
// This list is dynamically generated in preBuild
var CapabilityIds;
(function (CapabilityIds) {
    CapabilityIds["ACCOUNTPROFILE"] = "accountProfile";
    CapabilityIds["ACCOUNTSEGMENTS"] = "accountSegments";
    CapabilityIds["ADMIN"] = "admin";
    CapabilityIds["ALERT"] = "alert";
    CapabilityIds["APPCONFIGURATIONS"] = "appconfigurations";
    CapabilityIds["APPLICATIONS"] = "applications";
    CapabilityIds["APPSURFACES"] = "appsurfaces";
    CapabilityIds["ASSETS"] = "assets";
    CapabilityIds["ASSURANCE"] = "assurance";
    CapabilityIds["CAMPAIGNS"] = "campaigns";
    CapabilityIds["CLIENTPROPERTIES"] = "clientproperties";
    CapabilityIds["COMPONENTS"] = "components";
    CapabilityIds["CONTENTTEMPLATES"] = "contentTemplates";
    CapabilityIds["CONTROLRULES"] = "controlRules";
    CapabilityIds["DASHBOARD"] = "dashboard";
    CapabilityIds["DATACLEANSING"] = "dataCleansing";
    CapabilityIds["DATACOLLECTIONMONITORING"] = "datacollectionmonitoring";
    CapabilityIds["DATAFLOW"] = "dataflow";
    CapabilityIds["DATASET"] = "dataset";
    CapabilityIds["DATASTREAMS"] = "datastreams";
    CapabilityIds["DATASTREAMSBETA"] = "datastreamsbeta";
    CapabilityIds["DESTINATION"] = "destination";
    CapabilityIds["FRAGMENTS"] = "fragments";
    CapabilityIds["HOME"] = "home";
    CapabilityIds["IDENTITIES"] = "identities";
    CapabilityIds["IDENTITY"] = "identity";
    CapabilityIds["INITIATIVES"] = "initiatives";
    CapabilityIds["JOURNEYS"] = "journeys";
    CapabilityIds["LANDINGPAGES"] = "landingPages";
    CapabilityIds["LICENSEUSAGE"] = "licenseUsage";
    CapabilityIds["MESSAGES"] = "messages";
    CapabilityIds["MODEL"] = "model";
    CapabilityIds["MONITORING"] = "monitoring";
    CapabilityIds["NOTEBOOK"] = "notebook";
    CapabilityIds["OFFER"] = "offer";
    CapabilityIds["OFFERACTIVITY"] = "offeractivity";
    CapabilityIds["PERMISSIONSUI"] = "permissionsUI";
    CapabilityIds["PLACES"] = "places";
    CapabilityIds["PLAYBOOKS"] = "playbooks";
    CapabilityIds["POLICY"] = "policy";
    CapabilityIds["PRIVACYAUDIT"] = "privacyAudit";
    CapabilityIds["PRIVACYCONSOLE"] = "privacyConsole";
    CapabilityIds["PROFILE"] = "profile";
    CapabilityIds["QUERY"] = "query";
    CapabilityIds["REQUEST"] = "request";
    CapabilityIds["SANDBOX"] = "sandbox";
    CapabilityIds["SCHEMA"] = "schema";
    CapabilityIds["SEGMENT"] = "segment";
    CapabilityIds["SERVERPROPERTIES"] = "serverproperties";
    CapabilityIds["SERVICE"] = "service";
    CapabilityIds["SETTINGS"] = "settings";
    CapabilityIds["SOURCE"] = "source";
    CapabilityIds["SUBSCRIPTIONS"] = "subscriptions";
    CapabilityIds["UNIFIEDTAGS"] = "unifiedTags";
    CapabilityIds["WORKFLOW"] = "workflow";
})(CapabilityIds = exports.CapabilityIds || (exports.CapabilityIds = {}));
const capabilityapi = {
    get: params => {
        return (0, Global_1.getImpl)('capabilityApi')().get(params);
    }
};
exports.default = capabilityapi;
