// @react-spectrum/divider/dist/main.js
require("./main.css");
var $7uVzT$reactspectrumutils = require("@react-spectrum/utils");
var $7uVzT$react = require("react");
var $7uVzT$reactariaseparator = require("@react-aria/separator");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Divider", () => $a75f934b8a5939c6$export$2e0a83ec2e27ecbb);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 

var $7fdcb1fd1927dca1$exports = {};

$parcel$export($7fdcb1fd1927dca1$exports, "spectrum-Rule", () => $7fdcb1fd1927dca1$export$9ebf9cdaed6878bc, (v) => $7fdcb1fd1927dca1$export$9ebf9cdaed6878bc = v);
$parcel$export($7fdcb1fd1927dca1$exports, "spectrum-Rule--large", () => $7fdcb1fd1927dca1$export$243baa0ec4ab58cc, (v) => $7fdcb1fd1927dca1$export$243baa0ec4ab58cc = v);
$parcel$export($7fdcb1fd1927dca1$exports, "spectrum-Rule--medium", () => $7fdcb1fd1927dca1$export$a8d73b58ca350eb9, (v) => $7fdcb1fd1927dca1$export$a8d73b58ca350eb9 = v);
$parcel$export($7fdcb1fd1927dca1$exports, "spectrum-Rule--small", () => $7fdcb1fd1927dca1$export$b4b727880cfe37d, (v) => $7fdcb1fd1927dca1$export$b4b727880cfe37d = v);
$parcel$export($7fdcb1fd1927dca1$exports, "spectrum-Rule--horizontal", () => $7fdcb1fd1927dca1$export$563c1b24466c05b8, (v) => $7fdcb1fd1927dca1$export$563c1b24466c05b8 = v);
$parcel$export($7fdcb1fd1927dca1$exports, "spectrum-Rule--vertical", () => $7fdcb1fd1927dca1$export$76f93463a8fc15a8, (v) => $7fdcb1fd1927dca1$export$76f93463a8fc15a8 = v);
var $7fdcb1fd1927dca1$export$9ebf9cdaed6878bc;
var $7fdcb1fd1927dca1$export$243baa0ec4ab58cc;
var $7fdcb1fd1927dca1$export$a8d73b58ca350eb9;
var $7fdcb1fd1927dca1$export$b4b727880cfe37d;
var $7fdcb1fd1927dca1$export$563c1b24466c05b8;
var $7fdcb1fd1927dca1$export$76f93463a8fc15a8;
$7fdcb1fd1927dca1$export$9ebf9cdaed6878bc = "spectrum-Rule_612bb8";
$7fdcb1fd1927dca1$export$243baa0ec4ab58cc = "spectrum-Rule--large_612bb8";
$7fdcb1fd1927dca1$export$a8d73b58ca350eb9 = "spectrum-Rule--medium_612bb8";
$7fdcb1fd1927dca1$export$b4b727880cfe37d = "spectrum-Rule--small_612bb8";
$7fdcb1fd1927dca1$export$563c1b24466c05b8 = "spectrum-Rule--horizontal_612bb8";
$7fdcb1fd1927dca1$export$76f93463a8fc15a8 = "spectrum-Rule--vertical_612bb8";



let $a75f934b8a5939c6$var$sizeMap = {
    S: "small",
    M: "medium",
    L: "large"
};
function $a75f934b8a5939c6$var$Divider(props, ref) {
    props = (0, $7uVzT$reactspectrumutils.useSlotProps)(props, "divider");
    let { size: size = "L" , orientation: orientation = "horizontal" , ...otherProps } = props;
    let domRef = (0, $7uVzT$reactspectrumutils.useDOMRef)(ref);
    let { styleProps: styleProps  } = (0, $7uVzT$reactspectrumutils.useStyleProps)(otherProps);
    let weight = $a75f934b8a5939c6$var$sizeMap[size];
    let Element = "hr";
    if (orientation === "vertical") Element = "div";
    let { separatorProps: separatorProps  } = (0, $7uVzT$reactariaseparator.useSeparator)({
        ...props,
        elementType: Element
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($7uVzT$react))).createElement(Element, {
        ...styleProps,
        className: (0, $7uVzT$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($7fdcb1fd1927dca1$exports))), "spectrum-Rule", `spectrum-Rule--${weight}`, {
            "spectrum-Rule--vertical": orientation === "vertical",
            "spectrum-Rule--horizontal": orientation === "horizontal"
        }, styleProps.className),
        // @ts-ignore https://github.com/Microsoft/TypeScript/issues/28892
        ref: domRef,
        ...separatorProps
    });
}
/**
 * Dividers bring clarity to a layout by grouping and dividing content in close proximity.
 * They can also be used to establish rhythm and hierarchy.
 */ let $a75f934b8a5939c6$export$2e0a83ec2e27ecbb = /*#__PURE__*/ (0, ($parcel$interopDefault($7uVzT$react))).forwardRef($a75f934b8a5939c6$var$Divider);
