// @react-spectrum/searchfield/dist/main.js
require("./main.css");
var $kYnKt$reactspectrumutils = require("@react-spectrum/utils");
var $kYnKt$reactspectrumbutton = require("@react-spectrum/button");
var $kYnKt$spectrumiconsuiMagnifier = require("@spectrum-icons/ui/Magnifier");
var $kYnKt$react = require("react");
var $kYnKt$reactspectrumtextfield = require("@react-spectrum/textfield");
var $kYnKt$reactspectrumprovider = require("@react-spectrum/provider");
var $kYnKt$reactariasearchfield = require("@react-aria/searchfield");
var $kYnKt$reactstatelysearchfield = require("@react-stately/searchfield");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "SearchField", () => $111fe330b4c1f3ac$export$b94867ecbd698f21);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



var $1e139f06a0a4b696$exports = {};

$parcel$export($1e139f06a0a4b696$exports, "spectrum-Search", () => $1e139f06a0a4b696$export$713911d754e1e4ae, (v) => $1e139f06a0a4b696$export$713911d754e1e4ae = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-ClearButton", () => $1e139f06a0a4b696$export$b752ce409e5660c, (v) => $1e139f06a0a4b696$export$b752ce409e5660c = v);
$parcel$export($1e139f06a0a4b696$exports, "is-quiet", () => $1e139f06a0a4b696$export$6d5c5433f0da4241, (v) => $1e139f06a0a4b696$export$6d5c5433f0da4241 = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Search-input", () => $1e139f06a0a4b696$export$50e02603f2d97ae1, (v) => $1e139f06a0a4b696$export$50e02603f2d97ae1 = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Search--invalid", () => $1e139f06a0a4b696$export$77a6151a0b46a6a3, (v) => $1e139f06a0a4b696$export$77a6151a0b46a6a3 = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Search--valid", () => $1e139f06a0a4b696$export$aa81e3ab40f37567, (v) => $1e139f06a0a4b696$export$aa81e3ab40f37567 = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Search--loadable", () => $1e139f06a0a4b696$export$9c74d64afae3b2d4, (v) => $1e139f06a0a4b696$export$9c74d64afae3b2d4 = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Search-circleLoader", () => $1e139f06a0a4b696$export$70ae4353602451dc, (v) => $1e139f06a0a4b696$export$70ae4353602451dc = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Search-validationIcon", () => $1e139f06a0a4b696$export$938f0981825eae6e, (v) => $1e139f06a0a4b696$export$938f0981825eae6e = v);
$parcel$export($1e139f06a0a4b696$exports, "spectrum-Textfield", () => $1e139f06a0a4b696$export$2c8a10299d8b3418, (v) => $1e139f06a0a4b696$export$2c8a10299d8b3418 = v);
var $1e139f06a0a4b696$export$713911d754e1e4ae;
var $1e139f06a0a4b696$export$b752ce409e5660c;
var $1e139f06a0a4b696$export$6d5c5433f0da4241;
var $1e139f06a0a4b696$export$50e02603f2d97ae1;
var $1e139f06a0a4b696$export$77a6151a0b46a6a3;
var $1e139f06a0a4b696$export$aa81e3ab40f37567;
var $1e139f06a0a4b696$export$9c74d64afae3b2d4;
var $1e139f06a0a4b696$export$70ae4353602451dc;
var $1e139f06a0a4b696$export$938f0981825eae6e;
var $1e139f06a0a4b696$export$2c8a10299d8b3418;
$1e139f06a0a4b696$export$713911d754e1e4ae = "spectrum-Search_d3c0f3";
$1e139f06a0a4b696$export$b752ce409e5660c = "spectrum-ClearButton_d3c0f3";
$1e139f06a0a4b696$export$6d5c5433f0da4241 = "is-quiet_d3c0f3";
$1e139f06a0a4b696$export$50e02603f2d97ae1 = "spectrum-Search-input_d3c0f3";
$1e139f06a0a4b696$export$77a6151a0b46a6a3 = "spectrum-Search--invalid_d3c0f3";
$1e139f06a0a4b696$export$aa81e3ab40f37567 = "spectrum-Search--valid_d3c0f3";
$1e139f06a0a4b696$export$9c74d64afae3b2d4 = "spectrum-Search--loadable_d3c0f3";
$1e139f06a0a4b696$export$70ae4353602451dc = "spectrum-Search-circleLoader_d3c0f3";
$1e139f06a0a4b696$export$938f0981825eae6e = "spectrum-Search-validationIcon_d3c0f3";
$1e139f06a0a4b696$export$2c8a10299d8b3418 = "spectrum-Textfield_d3c0f3";






function $111fe330b4c1f3ac$var$SearchField(props, ref) {
    props = (0, $kYnKt$reactspectrumutils.useSlotProps)(props, "searchfield");
    props = (0, $kYnKt$reactspectrumprovider.useProviderProps)(props);
    let defaultIcon = /*#__PURE__*/ (0, ($parcel$interopDefault($kYnKt$react))).createElement((0, ($parcel$interopDefault($kYnKt$spectrumiconsuiMagnifier))), {
        "data-testid": "searchicon"
    });
    let { icon: icon = defaultIcon , isDisabled: isDisabled , UNSAFE_className: UNSAFE_className , placeholder: placeholder , ...otherProps } = props;
    if (placeholder) console.warn("Placeholders are deprecated due to accessibility issues. Please use help text instead. See the docs for details: https://react-spectrum.adobe.com/react-spectrum/SearchField.html#help-text");
    let state = (0, $kYnKt$reactstatelysearchfield.useSearchFieldState)(props);
    let inputRef = (0, $kYnKt$react.useRef)();
    let { labelProps: labelProps , inputProps: inputProps , clearButtonProps: clearButtonProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = (0, $kYnKt$reactariasearchfield.useSearchField)(props, state, inputRef);
    let clearButton = /*#__PURE__*/ (0, ($parcel$interopDefault($kYnKt$react))).createElement((0, $kYnKt$reactspectrumbutton.ClearButton), {
        ...clearButtonProps,
        preventFocus: true,
        UNSAFE_className: (0, $kYnKt$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($1e139f06a0a4b696$exports))), "spectrum-ClearButton"),
        isDisabled: isDisabled
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($kYnKt$react))).createElement((0, $kYnKt$reactspectrumtextfield.TextFieldBase), {
        ...otherProps,
        labelProps: labelProps,
        inputProps: inputProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        UNSAFE_className: (0, $kYnKt$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($1e139f06a0a4b696$exports))), "spectrum-Search", "spectrum-Textfield", {
            "is-disabled": isDisabled,
            "is-quiet": props.isQuiet,
            "spectrum-Search--invalid": props.validationState === "invalid" && !isDisabled,
            "spectrum-Search--valid": props.validationState === "valid" && !isDisabled
        }, UNSAFE_className),
        inputClassName: (0, $kYnKt$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($1e139f06a0a4b696$exports))), "spectrum-Search-input"),
        ref: ref,
        inputRef: inputRef,
        isDisabled: isDisabled,
        icon: icon,
        wrapperChildren: state.value !== "" && !props.isReadOnly && clearButton
    });
}
/**
 * A SearchField is a text field designed for searches.
 */ let $111fe330b4c1f3ac$export$b94867ecbd698f21 = /*#__PURE__*/ (0, $kYnKt$react.forwardRef)($111fe330b4c1f3ac$var$SearchField);
