// @react-spectrum/labeledvalue/dist/main.js
require("./main.css");
var $3cJLQ$internationalizeddate = require("@internationalized/date");
var $3cJLQ$reactspectrumutils = require("@react-spectrum/utils");
var $3cJLQ$reactspectrumlabel = require("@react-spectrum/label");
var $3cJLQ$reactariautils = require("@react-aria/utils");
var $3cJLQ$react = require("react");
var $3cJLQ$reactariai18n = require("@react-aria/i18n");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "LabeledValue", () => $2a7417c2df2197f7$export$d1328f67a56fa517);
/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



var $eb2fc0ee655eff6b$exports = {};

$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-FieldLabel", () => $eb2fc0ee655eff6b$export$493d18e796ae054, (v) => $eb2fc0ee655eff6b$export$493d18e796ae054 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-FieldLabel--positionSide", () => $eb2fc0ee655eff6b$export$9e6f19dc21f22f2e, (v) => $eb2fc0ee655eff6b$export$9e6f19dc21f22f2e = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-FieldLabel-requiredIcon", () => $eb2fc0ee655eff6b$export$7c47642c7d46f3c9, (v) => $eb2fc0ee655eff6b$export$7c47642c7d46f3c9 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-FieldLabel--alignEnd", () => $eb2fc0ee655eff6b$export$885efcc08143a987, (v) => $eb2fc0ee655eff6b$export$885efcc08143a987 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field", () => $eb2fc0ee655eff6b$export$f6d480ae1e56eba0, (v) => $eb2fc0ee655eff6b$export$f6d480ae1e56eba0 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field-contextualHelp", () => $eb2fc0ee655eff6b$export$24fb860d16dd2d3f, (v) => $eb2fc0ee655eff6b$export$24fb860d16dd2d3f = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field--positionTop", () => $eb2fc0ee655eff6b$export$a4ea780a9064d7f9, (v) => $eb2fc0ee655eff6b$export$a4ea780a9064d7f9 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field-field", () => $eb2fc0ee655eff6b$export$3ab8a3dc7f0563d, (v) => $eb2fc0ee655eff6b$export$3ab8a3dc7f0563d = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field--alignEnd", () => $eb2fc0ee655eff6b$export$5354c9be7ada5f83, (v) => $eb2fc0ee655eff6b$export$5354c9be7ada5f83 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field--positionSide", () => $eb2fc0ee655eff6b$export$2963225c91179589, (v) => $eb2fc0ee655eff6b$export$2963225c91179589 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field-wrapper", () => $eb2fc0ee655eff6b$export$127df0b7290eb8ef, (v) => $eb2fc0ee655eff6b$export$127df0b7290eb8ef = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field--hasContextualHelp", () => $eb2fc0ee655eff6b$export$9ba23707a252cdd4, (v) => $eb2fc0ee655eff6b$export$9ba23707a252cdd4 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Form", () => $eb2fc0ee655eff6b$export$94d2bcc94afabd89, (v) => $eb2fc0ee655eff6b$export$94d2bcc94afabd89 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Form--positionSide", () => $eb2fc0ee655eff6b$export$7f44db659563c8f4, (v) => $eb2fc0ee655eff6b$export$7f44db659563c8f4 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field-labelCell", () => $eb2fc0ee655eff6b$export$9cde5aecff012641, (v) => $eb2fc0ee655eff6b$export$9cde5aecff012641 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Field-labelWrapper", () => $eb2fc0ee655eff6b$export$a872a8c78b26c9e0, (v) => $eb2fc0ee655eff6b$export$a872a8c78b26c9e0 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Form--positionTop", () => $eb2fc0ee655eff6b$export$11b8bc81fe551057, (v) => $eb2fc0ee655eff6b$export$11b8bc81fe551057 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-LabeledValue", () => $eb2fc0ee655eff6b$export$79f9a5e9dd010c8d, (v) => $eb2fc0ee655eff6b$export$79f9a5e9dd010c8d = v);
$parcel$export($eb2fc0ee655eff6b$exports, "spectrum-Form-itemLabel", () => $eb2fc0ee655eff6b$export$51ec27881a1e3928, (v) => $eb2fc0ee655eff6b$export$51ec27881a1e3928 = v);
$parcel$export($eb2fc0ee655eff6b$exports, "is-disabled", () => $eb2fc0ee655eff6b$export$d35bc1e505d1ebbf, (v) => $eb2fc0ee655eff6b$export$d35bc1e505d1ebbf = v);
var $eb2fc0ee655eff6b$export$493d18e796ae054;
var $eb2fc0ee655eff6b$export$9e6f19dc21f22f2e;
var $eb2fc0ee655eff6b$export$7c47642c7d46f3c9;
var $eb2fc0ee655eff6b$export$885efcc08143a987;
var $eb2fc0ee655eff6b$export$f6d480ae1e56eba0;
var $eb2fc0ee655eff6b$export$24fb860d16dd2d3f;
var $eb2fc0ee655eff6b$export$a4ea780a9064d7f9;
var $eb2fc0ee655eff6b$export$3ab8a3dc7f0563d;
var $eb2fc0ee655eff6b$export$5354c9be7ada5f83;
var $eb2fc0ee655eff6b$export$2963225c91179589;
var $eb2fc0ee655eff6b$export$127df0b7290eb8ef;
var $eb2fc0ee655eff6b$export$9ba23707a252cdd4;
var $eb2fc0ee655eff6b$export$94d2bcc94afabd89;
var $eb2fc0ee655eff6b$export$7f44db659563c8f4;
var $eb2fc0ee655eff6b$export$9cde5aecff012641;
var $eb2fc0ee655eff6b$export$a872a8c78b26c9e0;
var $eb2fc0ee655eff6b$export$11b8bc81fe551057;
var $eb2fc0ee655eff6b$export$79f9a5e9dd010c8d;
var $eb2fc0ee655eff6b$export$51ec27881a1e3928;
var $eb2fc0ee655eff6b$export$d35bc1e505d1ebbf;
$eb2fc0ee655eff6b$export$493d18e796ae054 = "spectrum-FieldLabel_d2db1f";
$eb2fc0ee655eff6b$export$9e6f19dc21f22f2e = "spectrum-FieldLabel--positionSide_d2db1f";
$eb2fc0ee655eff6b$export$7c47642c7d46f3c9 = "spectrum-FieldLabel-requiredIcon_d2db1f";
$eb2fc0ee655eff6b$export$885efcc08143a987 = "spectrum-FieldLabel--alignEnd_d2db1f";
$eb2fc0ee655eff6b$export$f6d480ae1e56eba0 = "spectrum-Field_d2db1f";
$eb2fc0ee655eff6b$export$24fb860d16dd2d3f = "spectrum-Field-contextualHelp_d2db1f";
$eb2fc0ee655eff6b$export$a4ea780a9064d7f9 = "spectrum-Field--positionTop_d2db1f";
$eb2fc0ee655eff6b$export$3ab8a3dc7f0563d = "spectrum-Field-field_d2db1f";
$eb2fc0ee655eff6b$export$5354c9be7ada5f83 = "spectrum-Field--alignEnd_d2db1f";
$eb2fc0ee655eff6b$export$2963225c91179589 = "spectrum-Field--positionSide_d2db1f";
$eb2fc0ee655eff6b$export$127df0b7290eb8ef = "spectrum-Field-wrapper_d2db1f";
$eb2fc0ee655eff6b$export$9ba23707a252cdd4 = "spectrum-Field--hasContextualHelp_d2db1f";
$eb2fc0ee655eff6b$export$94d2bcc94afabd89 = "spectrum-Form_d2db1f";
$eb2fc0ee655eff6b$export$7f44db659563c8f4 = "spectrum-Form--positionSide_d2db1f";
$eb2fc0ee655eff6b$export$9cde5aecff012641 = "spectrum-Field-labelCell_d2db1f";
$eb2fc0ee655eff6b$export$a872a8c78b26c9e0 = "spectrum-Field-labelWrapper_d2db1f";
$eb2fc0ee655eff6b$export$11b8bc81fe551057 = "spectrum-Form--positionTop_d2db1f";
$eb2fc0ee655eff6b$export$79f9a5e9dd010c8d = "spectrum-LabeledValue_d2db1f";
$eb2fc0ee655eff6b$export$51ec27881a1e3928 = "spectrum-Form-itemLabel_d2db1f";
$eb2fc0ee655eff6b$export$d35bc1e505d1ebbf = "is-disabled_d2db1f";




function $2a7417c2df2197f7$var$LabeledValue(props, ref) {
    let { value: value , formatOptions: formatOptions  } = props;
    let domRef = (0, $3cJLQ$reactspectrumutils.useDOMRef)(ref);
    let children;
    if (Array.isArray(value)) // @ts-ignore
    children = /*#__PURE__*/ (0, ($parcel$interopDefault($3cJLQ$react))).createElement($2a7417c2df2197f7$var$FormattedStringList, {
        value: value,
        formatOptions: formatOptions
    });
    if (typeof value === "object" && "start" in value && typeof value.start === "number" && typeof value.end === "number") children = /*#__PURE__*/ (0, ($parcel$interopDefault($3cJLQ$react))).createElement($2a7417c2df2197f7$var$FormattedNumber, {
        value: value,
        formatOptions: formatOptions
    });
    if (typeof value === "object" && "start" in value && typeof value.start !== "number" && typeof value.end !== "number") children = /*#__PURE__*/ (0, ($parcel$interopDefault($3cJLQ$react))).createElement($2a7417c2df2197f7$var$FormattedDate, {
        value: value,
        formatOptions: formatOptions
    });
    if (typeof value === "number") children = /*#__PURE__*/ (0, ($parcel$interopDefault($3cJLQ$react))).createElement($2a7417c2df2197f7$var$FormattedNumber, {
        value: value,
        formatOptions: formatOptions
    });
    if (typeof value === "object" && ("calendar" in value || "hour" in value) || value instanceof Date) children = /*#__PURE__*/ (0, ($parcel$interopDefault($3cJLQ$react))).createElement($2a7417c2df2197f7$var$FormattedDate, {
        value: value,
        formatOptions: formatOptions
    });
    if (typeof value === "string") children = value;
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3cJLQ$react))).createElement((0, $3cJLQ$reactspectrumlabel.Field), {
        ...props,
        wrapperProps: (0, $3cJLQ$reactariautils.filterDOMProps)(props),
        ref: domRef,
        elementType: "span",
        wrapperClassName: (0, $3cJLQ$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($eb2fc0ee655eff6b$exports))), "spectrum-LabeledValue")
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($3cJLQ$react))).createElement("span", null, children));
}
function $2a7417c2df2197f7$var$FormattedStringList(props) {
    let stringFormatter = (0, $3cJLQ$reactariai18n.useListFormatter)(props.formatOptions);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3cJLQ$react))).createElement((0, ($parcel$interopDefault($3cJLQ$react))).Fragment, null, stringFormatter.format(props.value));
}
function $2a7417c2df2197f7$var$FormattedNumber(props) {
    let numberFormatter = (0, $3cJLQ$reactariai18n.useNumberFormatter)(props.formatOptions);
    let value = props.value;
    if (typeof value === "object") return /*#__PURE__*/ (0, ($parcel$interopDefault($3cJLQ$react))).createElement((0, ($parcel$interopDefault($3cJLQ$react))).Fragment, null, numberFormatter.formatRange(value.start, value.end));
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3cJLQ$react))).createElement((0, ($parcel$interopDefault($3cJLQ$react))).Fragment, null, numberFormatter.format(value));
}
function $2a7417c2df2197f7$var$FormattedDate(props) {
    let { value: value , formatOptions: formatOptions  } = props;
    if (!formatOptions) formatOptions = $2a7417c2df2197f7$var$getDefaultFormatOptions("start" in value ? value.start : value);
    let dateFormatter = (0, $3cJLQ$reactariai18n.useDateFormatter)(formatOptions);
    let timeZone = dateFormatter.resolvedOptions().timeZone || (0, $3cJLQ$internationalizeddate.getLocalTimeZone)();
    let final;
    if ("start" in value && "end" in value) {
        let start = value.start;
        let end = value.end;
        start = $2a7417c2df2197f7$var$convertDateTime(start, timeZone);
        end = $2a7417c2df2197f7$var$convertDateTime(end, timeZone);
        return /*#__PURE__*/ (0, ($parcel$interopDefault($3cJLQ$react))).createElement((0, ($parcel$interopDefault($3cJLQ$react))).Fragment, null, dateFormatter.formatRange(start, end));
    }
    final = $2a7417c2df2197f7$var$convertDateTime(value, timeZone);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($3cJLQ$react))).createElement((0, ($parcel$interopDefault($3cJLQ$react))).Fragment, null, dateFormatter.format(final));
}
function $2a7417c2df2197f7$var$convertDateTime(value, timeZone) {
    if ("timeZone" in value) return value.toDate();
    else if ("calendar" in value) return value.toDate(timeZone);
    else if (!(value instanceof Date)) return $2a7417c2df2197f7$var$convertValue(value).toDate(timeZone);
    return value;
}
function $2a7417c2df2197f7$var$getDefaultFormatOptions(value) {
    if (value instanceof Date) return {
        dateStyle: "long",
        timeStyle: "short"
    };
    else if ("timeZone" in value) return {
        year: "numeric",
        month: "long",
        day: "numeric",
        hour: "numeric",
        minute: "numeric",
        timeZone: value.timeZone,
        timeZoneName: "short"
    };
    else if ("hour" in value && "year" in value) return {
        dateStyle: "long",
        timeStyle: "short"
    };
    else if ("hour" in value) return {
        timeStyle: "short"
    };
    else return {
        dateStyle: "long"
    };
}
function $2a7417c2df2197f7$var$convertValue(value) {
    let date = (0, $3cJLQ$internationalizeddate.today)((0, $3cJLQ$internationalizeddate.getLocalTimeZone)());
    return (0, $3cJLQ$internationalizeddate.toCalendarDateTime)(date, value);
}
/**
 * A LabeledValue displays a non-editable value with a label. It formats numbers, dates, times, and lists according to the user's locale.
 */ let $2a7417c2df2197f7$export$d1328f67a56fa517 = /*#__PURE__*/ (0, ($parcel$interopDefault($3cJLQ$react))).forwardRef($2a7417c2df2197f7$var$LabeledValue);
