// @react-spectrum/radio/dist/main.js
require("./main.css");
var $ltKMc$reactspectrumutils = require("@react-spectrum/utils");
var $ltKMc$reactariafocus = require("@react-aria/focus");
var $ltKMc$react = require("react");
var $ltKMc$reactariainteractions = require("@react-aria/interactions");
var $ltKMc$reactariaradio = require("@react-aria/radio");
var $ltKMc$reactspectrumlabel = require("@react-spectrum/label");
var $ltKMc$reactspectrumform = require("@react-spectrum/form");
var $ltKMc$reactspectrumprovider = require("@react-spectrum/provider");
var $ltKMc$reactstatelyradio = require("@react-stately/radio");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Radio", () => $e8d5518f9c1aca76$export$d7b12c4107be0d61);
$parcel$export(module.exports, "RadioGroup", () => $aa5529d97b69d048$export$a98f0dcb43a68a25);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $a8fbff4889c18df2$exports = {};

$parcel$export($a8fbff4889c18df2$exports, "spectrum-Radio", () => $a8fbff4889c18df2$export$26ebd4dbfef068c5, (v) => $a8fbff4889c18df2$export$26ebd4dbfef068c5 = v);
$parcel$export($a8fbff4889c18df2$exports, "spectrum-Radio-input", () => $a8fbff4889c18df2$export$c0ce6f9a6a2de1cc, (v) => $a8fbff4889c18df2$export$c0ce6f9a6a2de1cc = v);
$parcel$export($a8fbff4889c18df2$exports, "spectrum-Radio-button", () => $a8fbff4889c18df2$export$bf9d5e902c33bf05, (v) => $a8fbff4889c18df2$export$bf9d5e902c33bf05 = v);
$parcel$export($a8fbff4889c18df2$exports, "focus-ring", () => $a8fbff4889c18df2$export$f39a09f249340e2a, (v) => $a8fbff4889c18df2$export$f39a09f249340e2a = v);
$parcel$export($a8fbff4889c18df2$exports, "spectrum-Radio-label", () => $a8fbff4889c18df2$export$b0f8395a9667922d, (v) => $a8fbff4889c18df2$export$b0f8395a9667922d = v);
$parcel$export($a8fbff4889c18df2$exports, "spectrum-Radio--labelBelow", () => $a8fbff4889c18df2$export$9443e7082698950d, (v) => $a8fbff4889c18df2$export$9443e7082698950d = v);
$parcel$export($a8fbff4889c18df2$exports, "is-hovered", () => $a8fbff4889c18df2$export$b8813cd5d7824ce7, (v) => $a8fbff4889c18df2$export$b8813cd5d7824ce7 = v);
$parcel$export($a8fbff4889c18df2$exports, "spectrum-Radio--quiet", () => $a8fbff4889c18df2$export$6db66b55a897f36d, (v) => $a8fbff4889c18df2$export$6db66b55a897f36d = v);
$parcel$export($a8fbff4889c18df2$exports, "is-invalid", () => $a8fbff4889c18df2$export$14a30de2866685fa, (v) => $a8fbff4889c18df2$export$14a30de2866685fa = v);
var $a8fbff4889c18df2$export$26ebd4dbfef068c5;
var $a8fbff4889c18df2$export$c0ce6f9a6a2de1cc;
var $a8fbff4889c18df2$export$bf9d5e902c33bf05;
var $a8fbff4889c18df2$export$f39a09f249340e2a;
var $a8fbff4889c18df2$export$b0f8395a9667922d;
var $a8fbff4889c18df2$export$9443e7082698950d;
var $a8fbff4889c18df2$export$b8813cd5d7824ce7;
var $a8fbff4889c18df2$export$6db66b55a897f36d;
var $a8fbff4889c18df2$export$14a30de2866685fa;
$a8fbff4889c18df2$export$26ebd4dbfef068c5 = "spectrum-Radio_bf46c2";
$a8fbff4889c18df2$export$c0ce6f9a6a2de1cc = "spectrum-Radio-input_bf46c2";
$a8fbff4889c18df2$export$bf9d5e902c33bf05 = "spectrum-Radio-button_bf46c2";
$a8fbff4889c18df2$export$f39a09f249340e2a = "focus-ring_bf46c2";
$a8fbff4889c18df2$export$b0f8395a9667922d = "spectrum-Radio-label_bf46c2";
$a8fbff4889c18df2$export$9443e7082698950d = "spectrum-Radio--labelBelow_bf46c2";
$a8fbff4889c18df2$export$b8813cd5d7824ce7 = "is-hovered_bf46c2";
$a8fbff4889c18df2$export$6db66b55a897f36d = "spectrum-Radio--quiet_bf46c2";
$a8fbff4889c18df2$export$14a30de2866685fa = "is-invalid_bf46c2";




/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
const $f5a8721b9af05d97$export$b118023277d4a5c3 = (0, ($parcel$interopDefault($ltKMc$react))).createContext(null);
function $f5a8721b9af05d97$export$b054eba74077a826() {
    return (0, $ltKMc$react.useContext)($f5a8721b9af05d97$export$b118023277d4a5c3);
}


function $e8d5518f9c1aca76$var$Radio(props, ref) {
    let { isDisabled: isDisabled , children: children , autoFocus: autoFocus , ...otherProps } = props;
    let { styleProps: styleProps  } = (0, $ltKMc$reactspectrumutils.useStyleProps)(otherProps);
    let { hoverProps: hoverProps , isHovered: isHovered  } = (0, $ltKMc$reactariainteractions.useHover)({
        isDisabled: isDisabled
    });
    let inputRef = (0, $ltKMc$react.useRef)(null);
    let domRef = (0, $ltKMc$reactspectrumutils.useFocusableRef)(ref, inputRef);
    let radioGroupProps = (0, $f5a8721b9af05d97$export$b054eba74077a826)();
    let { isEmphasized: isEmphasized , validationState: validationState , state: state  } = radioGroupProps;
    let { inputProps: inputProps  } = (0, $ltKMc$reactariaradio.useRadio)({
        ...props,
        ...radioGroupProps,
        isDisabled: isDisabled
    }, state, inputRef);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($ltKMc$react))).createElement("label", {
        ...styleProps,
        ...hoverProps,
        ref: domRef,
        className: (0, $ltKMc$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a8fbff4889c18df2$exports))), "spectrum-Radio", {
            // Removing. Pending design feedback.
            // 'spectrum-Radio--labelBelow': labelPosition === 'bottom',
            "spectrum-Radio--quiet": !isEmphasized,
            "is-disabled": isDisabled,
            "is-invalid": validationState === "invalid",
            "is-hovered": isHovered
        }, styleProps.className)
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($ltKMc$react))).createElement((0, $ltKMc$reactariafocus.FocusRing), {
        focusRingClass: (0, $ltKMc$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a8fbff4889c18df2$exports))), "focus-ring"),
        autoFocus: autoFocus
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($ltKMc$react))).createElement("input", {
        ...inputProps,
        ref: inputRef,
        className: (0, $ltKMc$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a8fbff4889c18df2$exports))), "spectrum-Radio-input")
    })), /*#__PURE__*/ (0, ($parcel$interopDefault($ltKMc$react))).createElement("span", {
        className: (0, $ltKMc$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a8fbff4889c18df2$exports))), "spectrum-Radio-button")
    }), children && /*#__PURE__*/ (0, ($parcel$interopDefault($ltKMc$react))).createElement("span", {
        className: (0, $ltKMc$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($a8fbff4889c18df2$exports))), "spectrum-Radio-label")
    }, children));
}
/**
 * Radio buttons allow users to select a single option from a list of mutually exclusive options.
 * All possible options are exposed up front for users to compare.
 */ const $e8d5518f9c1aca76$export$d7b12c4107be0d61 = /*#__PURE__*/ (0, $ltKMc$react.forwardRef)($e8d5518f9c1aca76$var$Radio);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 



var $53124a3a0fd56fca$exports = {};

$parcel$export($53124a3a0fd56fca$exports, "spectrum-FieldGroup", () => $53124a3a0fd56fca$export$75eb16cbfc18279c, (v) => $53124a3a0fd56fca$export$75eb16cbfc18279c = v);
$parcel$export($53124a3a0fd56fca$exports, "spectrum-FieldGroup-group", () => $53124a3a0fd56fca$export$984c7bbbfc7ddffc, (v) => $53124a3a0fd56fca$export$984c7bbbfc7ddffc = v);
$parcel$export($53124a3a0fd56fca$exports, "spectrum-FieldGroup-group--horizontal", () => $53124a3a0fd56fca$export$1ad8646bcba21c0e, (v) => $53124a3a0fd56fca$export$1ad8646bcba21c0e = v);
var $53124a3a0fd56fca$export$75eb16cbfc18279c;
var $53124a3a0fd56fca$export$984c7bbbfc7ddffc;
var $53124a3a0fd56fca$export$1ad8646bcba21c0e;
$53124a3a0fd56fca$export$75eb16cbfc18279c = "spectrum-FieldGroup_41d350";
$53124a3a0fd56fca$export$984c7bbbfc7ddffc = "spectrum-FieldGroup-group_41d350";
$53124a3a0fd56fca$export$1ad8646bcba21c0e = "spectrum-FieldGroup-group--horizontal_41d350";






function $aa5529d97b69d048$var$RadioGroup(props, ref) {
    props = (0, $ltKMc$reactspectrumprovider.useProviderProps)(props);
    props = (0, $ltKMc$reactspectrumform.useFormProps)(props);
    let { isEmphasized: isEmphasized , validationState: validationState , children: children , orientation: orientation = "vertical"  } = props;
    let domRef = (0, $ltKMc$reactspectrumutils.useDOMRef)(ref);
    let state = (0, $ltKMc$reactstatelyradio.useRadioGroupState)(props);
    let { radioGroupProps: radioGroupProps , labelProps: labelProps , descriptionProps: descriptionProps , errorMessageProps: errorMessageProps  } = (0, $ltKMc$reactariaradio.useRadioGroup)(props, state);
    return /*#__PURE__*/ (0, ($parcel$interopDefault($ltKMc$react))).createElement((0, $ltKMc$reactspectrumlabel.Field), {
        ...props,
        ref: domRef,
        wrapperClassName: (0, $ltKMc$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($53124a3a0fd56fca$exports))), "spectrum-FieldGroup"),
        labelProps: labelProps,
        descriptionProps: descriptionProps,
        errorMessageProps: errorMessageProps,
        elementType: "span"
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($ltKMc$react))).createElement("div", {
        ...radioGroupProps,
        className: (0, $ltKMc$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($53124a3a0fd56fca$exports))), "spectrum-FieldGroup-group", {
            "spectrum-FieldGroup-group--horizontal": orientation === "horizontal"
        })
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($ltKMc$react))).createElement((0, $f5a8721b9af05d97$export$b118023277d4a5c3).Provider, {
        value: {
            isEmphasized: isEmphasized,
            validationState: validationState,
            state: state
        }
    }, children)));
}
/**
 * Radio groups allow users to select a single option from a list of mutually exclusive options.
 * All possible options are exposed up front for users to compare.
 */ const $aa5529d97b69d048$export$a98f0dcb43a68a25 = /*#__PURE__*/ (0, ($parcel$interopDefault($ltKMc$react))).forwardRef($aa5529d97b69d048$var$RadioGroup);
