// @react-spectrum/tooltip/dist/main.js
require("./main.css");
var $kAvU8$spectrumiconsuiAlertSmall = require("@spectrum-icons/ui/AlertSmall");
var $kAvU8$reactspectrumutils = require("@react-spectrum/utils");
var $kAvU8$spectrumiconsuiInfoSmall = require("@spectrum-icons/ui/InfoSmall");
var $kAvU8$reactariautils = require("@react-aria/utils");
var $kAvU8$react = require("react");
var $kAvU8$spectrumiconsuiSuccessSmall = require("@spectrum-icons/ui/SuccessSmall");
var $kAvU8$reactariatooltip = require("@react-aria/tooltip");
var $kAvU8$reactariafocus = require("@react-aria/focus");
var $kAvU8$reactspectrumoverlays = require("@react-spectrum/overlays");
var $kAvU8$reactariaoverlays = require("@react-aria/overlays");
var $kAvU8$reactstatelytooltip = require("@react-stately/tooltip");

function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}

$parcel$export(module.exports, "Tooltip", () => $55bd11986fcffc6f$export$28c660c63b792dea);
$parcel$export(module.exports, "TooltipTrigger", () => $19f06480c1c1a932$export$8c610744efcf8a1d);
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 




var $ba73465d6d9b6556$exports = {};

$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip", () => $ba73465d6d9b6556$export$f7bf7078d3d03f05, (v) => $ba73465d6d9b6556$export$f7bf7078d3d03f05 = v);
$parcel$export($ba73465d6d9b6556$exports, "is-open", () => $ba73465d6d9b6556$export$a9781837241c946d, (v) => $ba73465d6d9b6556$export$a9781837241c946d = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--bottom", () => $ba73465d6d9b6556$export$4f420a9bbb442953, (v) => $ba73465d6d9b6556$export$4f420a9bbb442953 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--top", () => $ba73465d6d9b6556$export$d7516df59e198fc7, (v) => $ba73465d6d9b6556$export$d7516df59e198fc7 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--right", () => $ba73465d6d9b6556$export$c094187772823b2e, (v) => $ba73465d6d9b6556$export$c094187772823b2e = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--left", () => $ba73465d6d9b6556$export$47f060b68b4cab2c, (v) => $ba73465d6d9b6556$export$47f060b68b4cab2c = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip-tip", () => $ba73465d6d9b6556$export$9deb89a7a91de30, (v) => $ba73465d6d9b6556$export$9deb89a7a91de30 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip-typeIcon", () => $ba73465d6d9b6556$export$ff23d46fee8272e4, (v) => $ba73465d6d9b6556$export$ff23d46fee8272e4 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip-label", () => $ba73465d6d9b6556$export$f4096e0dc445b255, (v) => $ba73465d6d9b6556$export$f4096e0dc445b255 = v);
$parcel$export($ba73465d6d9b6556$exports, "u-tooltip-showOnHover", () => $ba73465d6d9b6556$export$6ea3237f62800d00, (v) => $ba73465d6d9b6556$export$6ea3237f62800d00 = v);
$parcel$export($ba73465d6d9b6556$exports, "is-hovered", () => $ba73465d6d9b6556$export$b8813cd5d7824ce7, (v) => $ba73465d6d9b6556$export$b8813cd5d7824ce7 = v);
$parcel$export($ba73465d6d9b6556$exports, "is-focused", () => $ba73465d6d9b6556$export$e7dc768d35940237, (v) => $ba73465d6d9b6556$export$e7dc768d35940237 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--negative", () => $ba73465d6d9b6556$export$92fe45b4260594f6, (v) => $ba73465d6d9b6556$export$92fe45b4260594f6 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--error", () => $ba73465d6d9b6556$export$2b291890a5577893, (v) => $ba73465d6d9b6556$export$2b291890a5577893 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--info", () => $ba73465d6d9b6556$export$e995d5780591d0a8, (v) => $ba73465d6d9b6556$export$e995d5780591d0a8 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--help", () => $ba73465d6d9b6556$export$d33ca445ae528241, (v) => $ba73465d6d9b6556$export$d33ca445ae528241 = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--positive", () => $ba73465d6d9b6556$export$11cbf31e1b6e51ac, (v) => $ba73465d6d9b6556$export$11cbf31e1b6e51ac = v);
$parcel$export($ba73465d6d9b6556$exports, "spectrum-Tooltip--success", () => $ba73465d6d9b6556$export$4eae622bc9b08cac, (v) => $ba73465d6d9b6556$export$4eae622bc9b08cac = v);
var $ba73465d6d9b6556$export$f7bf7078d3d03f05;
var $ba73465d6d9b6556$export$a9781837241c946d;
var $ba73465d6d9b6556$export$4f420a9bbb442953;
var $ba73465d6d9b6556$export$d7516df59e198fc7;
var $ba73465d6d9b6556$export$c094187772823b2e;
var $ba73465d6d9b6556$export$47f060b68b4cab2c;
var $ba73465d6d9b6556$export$9deb89a7a91de30;
var $ba73465d6d9b6556$export$ff23d46fee8272e4;
var $ba73465d6d9b6556$export$f4096e0dc445b255;
var $ba73465d6d9b6556$export$6ea3237f62800d00;
var $ba73465d6d9b6556$export$b8813cd5d7824ce7;
var $ba73465d6d9b6556$export$e7dc768d35940237;
var $ba73465d6d9b6556$export$92fe45b4260594f6;
var $ba73465d6d9b6556$export$2b291890a5577893;
var $ba73465d6d9b6556$export$e995d5780591d0a8;
var $ba73465d6d9b6556$export$d33ca445ae528241;
var $ba73465d6d9b6556$export$11cbf31e1b6e51ac;
var $ba73465d6d9b6556$export$4eae622bc9b08cac;
$ba73465d6d9b6556$export$f7bf7078d3d03f05 = "spectrum-Tooltip_22ae94";
$ba73465d6d9b6556$export$a9781837241c946d = "is-open_22ae94";
$ba73465d6d9b6556$export$4f420a9bbb442953 = "spectrum-Tooltip--bottom_22ae94";
$ba73465d6d9b6556$export$d7516df59e198fc7 = "spectrum-Tooltip--top_22ae94";
$ba73465d6d9b6556$export$c094187772823b2e = "spectrum-Tooltip--right_22ae94";
$ba73465d6d9b6556$export$47f060b68b4cab2c = "spectrum-Tooltip--left_22ae94";
$ba73465d6d9b6556$export$9deb89a7a91de30 = "spectrum-Tooltip-tip_22ae94";
$ba73465d6d9b6556$export$ff23d46fee8272e4 = "spectrum-Tooltip-typeIcon_22ae94";
$ba73465d6d9b6556$export$f4096e0dc445b255 = "spectrum-Tooltip-label_22ae94";
$ba73465d6d9b6556$export$6ea3237f62800d00 = "u-tooltip-showOnHover_22ae94";
$ba73465d6d9b6556$export$b8813cd5d7824ce7 = "is-hovered_22ae94";
$ba73465d6d9b6556$export$e7dc768d35940237 = "is-focused_22ae94";
$ba73465d6d9b6556$export$92fe45b4260594f6 = "spectrum-Tooltip--negative_22ae94";
$ba73465d6d9b6556$export$2b291890a5577893 = "spectrum-Tooltip--error_22ae94";
$ba73465d6d9b6556$export$e995d5780591d0a8 = "spectrum-Tooltip--info_22ae94";
$ba73465d6d9b6556$export$d33ca445ae528241 = "spectrum-Tooltip--help_22ae94";
$ba73465d6d9b6556$export$11cbf31e1b6e51ac = "spectrum-Tooltip--positive_22ae94";
$ba73465d6d9b6556$export$4eae622bc9b08cac = "spectrum-Tooltip--success_22ae94";



/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 
const $ed2e56c8503433b4$export$39ae08fa83328b12 = (0, ($parcel$interopDefault($kAvU8$react))).createContext({});



let $55bd11986fcffc6f$var$iconMap = {
    info: (0, ($parcel$interopDefault($kAvU8$spectrumiconsuiInfoSmall))),
    positive: (0, ($parcel$interopDefault($kAvU8$spectrumiconsuiSuccessSmall))),
    negative: (0, ($parcel$interopDefault($kAvU8$spectrumiconsuiAlertSmall)))
};
function $55bd11986fcffc6f$var$Tooltip(props, ref) {
    let { ref: overlayRef , arrowProps: arrowProps , state: state , ...tooltipProviderProps } = (0, $kAvU8$react.useContext)((0, $ed2e56c8503433b4$export$39ae08fa83328b12));
    let defaultRef = (0, $kAvU8$react.useRef)();
    overlayRef = overlayRef || defaultRef;
    props = (0, $kAvU8$reactariautils.mergeProps)(props, tooltipProviderProps);
    let { variant: variant = "neutral" , placement: placement = "top" , isOpen: isOpen , showIcon: showIcon , ...otherProps } = props;
    let { styleProps: styleProps  } = (0, $kAvU8$reactspectrumutils.useStyleProps)(otherProps);
    let { tooltipProps: tooltipProps  } = (0, $kAvU8$reactariatooltip.useTooltip)(props, state);
    // Sync ref with overlayRef from context.
    (0, $kAvU8$react.useImperativeHandle)(ref, ()=>(0, $kAvU8$reactspectrumutils.createDOMRef)(overlayRef));
    let Icon = $55bd11986fcffc6f$var$iconMap[variant];
    return /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement("div", {
        ...styleProps,
        ...tooltipProps,
        className: (0, $kAvU8$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba73465d6d9b6556$exports))), "spectrum-Tooltip", `spectrum-Tooltip--${variant}`, `spectrum-Tooltip--${placement}`, {
            "is-open": isOpen
        }, styleProps.className),
        ref: overlayRef
    }, showIcon && variant !== "neutral" && /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement(Icon, {
        UNSAFE_className: (0, $kAvU8$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba73465d6d9b6556$exports))), "spectrum-Tooltip-typeIcon"),
        "aria-hidden": true
    }), props.children && /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement("span", {
        className: (0, $kAvU8$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba73465d6d9b6556$exports))), "spectrum-Tooltip-label")
    }, props.children), /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement("span", {
        ...arrowProps,
        className: (0, $kAvU8$reactspectrumutils.classNames)((0, (/*@__PURE__*/$parcel$interopDefault($ba73465d6d9b6556$exports))), "spectrum-Tooltip-tip")
    }));
}
/**
 * Display container for Tooltip content. Has a directional arrow dependent on its placement.
 */ let $55bd11986fcffc6f$export$28c660c63b792dea = /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).forwardRef($55bd11986fcffc6f$var$Tooltip);


/*
 * Copyright 2020 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 






const $19f06480c1c1a932$var$DEFAULT_OFFSET = -1; // Offset needed to reach 4px/5px (med/large) distance between tooltip and trigger button
const $19f06480c1c1a932$var$DEFAULT_CROSS_OFFSET = 0;
function $19f06480c1c1a932$var$TooltipTrigger(props) {
    let { children: children , crossOffset: crossOffset = $19f06480c1c1a932$var$DEFAULT_CROSS_OFFSET , isDisabled: isDisabled , offset: offset = $19f06480c1c1a932$var$DEFAULT_OFFSET , trigger: triggerAction  } = props;
    let [trigger, tooltip] = (0, ($parcel$interopDefault($kAvU8$react))).Children.toArray(children);
    let state = (0, $kAvU8$reactstatelytooltip.useTooltipTriggerState)(props);
    let tooltipTriggerRef = (0, $kAvU8$react.useRef)();
    let overlayRef = (0, $kAvU8$react.useRef)();
    let { triggerProps: triggerProps , tooltipProps: tooltipProps  } = (0, $kAvU8$reactariatooltip.useTooltipTrigger)({
        isDisabled: isDisabled,
        trigger: triggerAction
    }, state, tooltipTriggerRef);
    let { overlayProps: overlayProps , arrowProps: arrowProps , placement: placement  } = (0, $kAvU8$reactariaoverlays.useOverlayPosition)({
        placement: props.placement || "top",
        targetRef: tooltipTriggerRef,
        overlayRef: overlayRef,
        offset: offset,
        crossOffset: crossOffset,
        isOpen: state.isOpen,
        shouldFlip: props.shouldFlip,
        containerPadding: props.containerPadding
    });
    return /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement((0, $kAvU8$reactariafocus.FocusableProvider), {
        ...triggerProps,
        ref: tooltipTriggerRef
    }, trigger, /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement((0, $ed2e56c8503433b4$export$39ae08fa83328b12).Provider, {
        value: {
            state: state,
            placement: placement,
            ref: overlayRef,
            UNSAFE_style: overlayProps.style,
            arrowProps: arrowProps,
            ...tooltipProps
        }
    }, /*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement((0, $kAvU8$reactspectrumoverlays.Overlay), {
        isOpen: state.isOpen
    }, tooltip)));
}
// Support TooltipTrigger inside components using CollectionBuilder.
$19f06480c1c1a932$var$TooltipTrigger.getCollectionNode = function*(props) {
    // Replaced the use of React.Childern.toArray because it mutates the key prop.
    let childArray = [];
    (0, ($parcel$interopDefault($kAvU8$react))).Children.forEach(props.children, (child)=>{
        if (/*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).isValidElement(child)) childArray.push(child);
    });
    let [trigger, tooltip] = childArray;
    yield {
        element: trigger,
        wrapper: (element)=>/*#__PURE__*/ (0, ($parcel$interopDefault($kAvU8$react))).createElement($19f06480c1c1a932$var$TooltipTrigger, {
                key: element.key,
                ...props
            }, element, tooltip)
    };
};
/**
 * TooltipTrigger wraps around a trigger element and a Tooltip. It handles opening and closing
 * the Tooltip when the user hovers over or focuses the trigger, and positioning the Tooltip
 * relative to the trigger.
 */ // We don't want getCollectionNode to show up in the type definition
let $19f06480c1c1a932$export$8c610744efcf8a1d = $19f06480c1c1a932$var$TooltipTrigger;
