// @react-spectrum/badge/dist/module.js
import "./main.css";
import {useDOMRef as $ha4zE$useDOMRef, useStyleProps as $ha4zE$useStyleProps, classNames as $ha4zE$classNames, SlotProvider as $ha4zE$SlotProvider} from "@react-spectrum/utils";
import {filterDOMProps as $ha4zE$filterDOMProps} from "@react-aria/utils";
import $ha4zE$react, {forwardRef as $ha4zE$forwardRef} from "react";
import {Text as $ha4zE$Text} from "@react-spectrum/text";
import {useProviderProps as $ha4zE$useProviderProps} from "@react-spectrum/provider";

function $parcel$interopDefault(a) {
  return a && a.__esModule ? a.default : a;
}
function $parcel$export(e, n, v, s) {
  Object.defineProperty(e, n, {get: v, set: s, enumerable: true, configurable: true});
}
/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ /// <reference types="css-module-types" />
/*
 * Copyright 2022 Adobe. All rights reserved.
 * This file is licensed to you under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License. You may obtain a copy
 * of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under
 * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 * OF ANY KIND, either express or implied. See the License for the specific language
 * governing permissions and limitations under the License.
 */ 


var $700c1b49452b6d47$exports = {};

$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge", () => $700c1b49452b6d47$export$acb311e43ff4d1, (v) => $700c1b49452b6d47$export$acb311e43ff4d1 = v);
$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge-label", () => $700c1b49452b6d47$export$87cc42d7dd800a50, (v) => $700c1b49452b6d47$export$87cc42d7dd800a50 = v);
$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge-icon", () => $700c1b49452b6d47$export$70ebbcb6132f4aa8, (v) => $700c1b49452b6d47$export$70ebbcb6132f4aa8 = v);
$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge--large", () => $700c1b49452b6d47$export$8e48598848736b6e, (v) => $700c1b49452b6d47$export$8e48598848736b6e = v);
$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge--small", () => $700c1b49452b6d47$export$e76d1f315203cda4, (v) => $700c1b49452b6d47$export$e76d1f315203cda4 = v);
$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge--neutral", () => $700c1b49452b6d47$export$ce0abe6a13066aee, (v) => $700c1b49452b6d47$export$ce0abe6a13066aee = v);
$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge--negative", () => $700c1b49452b6d47$export$250de4fc6219f4ba, (v) => $700c1b49452b6d47$export$250de4fc6219f4ba = v);
$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge--yellow", () => $700c1b49452b6d47$export$f39824621eafe7d1, (v) => $700c1b49452b6d47$export$f39824621eafe7d1 = v);
$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge--seafoam", () => $700c1b49452b6d47$export$b9072ec5c3c3196e, (v) => $700c1b49452b6d47$export$b9072ec5c3c3196e = v);
$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge--positive", () => $700c1b49452b6d47$export$9cb54bc972cacc06, (v) => $700c1b49452b6d47$export$9cb54bc972cacc06 = v);
$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge--info", () => $700c1b49452b6d47$export$a276b7cecb1e7608, (v) => $700c1b49452b6d47$export$a276b7cecb1e7608 = v);
$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge--fuchsia", () => $700c1b49452b6d47$export$17d8c34537918c7, (v) => $700c1b49452b6d47$export$17d8c34537918c7 = v);
$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge--indigo", () => $700c1b49452b6d47$export$cf21c81baa11bb0f, (v) => $700c1b49452b6d47$export$cf21c81baa11bb0f = v);
$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge--magenta", () => $700c1b49452b6d47$export$a6258e5c939933e, (v) => $700c1b49452b6d47$export$a6258e5c939933e = v);
$parcel$export($700c1b49452b6d47$exports, "spectrum-Badge--purple", () => $700c1b49452b6d47$export$67ff21277f017a3f, (v) => $700c1b49452b6d47$export$67ff21277f017a3f = v);
var $700c1b49452b6d47$export$acb311e43ff4d1;
var $700c1b49452b6d47$export$87cc42d7dd800a50;
var $700c1b49452b6d47$export$70ebbcb6132f4aa8;
var $700c1b49452b6d47$export$8e48598848736b6e;
var $700c1b49452b6d47$export$e76d1f315203cda4;
var $700c1b49452b6d47$export$ce0abe6a13066aee;
var $700c1b49452b6d47$export$250de4fc6219f4ba;
var $700c1b49452b6d47$export$f39824621eafe7d1;
var $700c1b49452b6d47$export$b9072ec5c3c3196e;
var $700c1b49452b6d47$export$9cb54bc972cacc06;
var $700c1b49452b6d47$export$a276b7cecb1e7608;
var $700c1b49452b6d47$export$17d8c34537918c7;
var $700c1b49452b6d47$export$cf21c81baa11bb0f;
var $700c1b49452b6d47$export$a6258e5c939933e;
var $700c1b49452b6d47$export$67ff21277f017a3f;
$700c1b49452b6d47$export$acb311e43ff4d1 = "spectrum-Badge_a70f1a";
$700c1b49452b6d47$export$87cc42d7dd800a50 = "spectrum-Badge-label_a70f1a";
$700c1b49452b6d47$export$70ebbcb6132f4aa8 = "spectrum-Badge-icon_a70f1a";
$700c1b49452b6d47$export$8e48598848736b6e = "spectrum-Badge--large_a70f1a";
$700c1b49452b6d47$export$e76d1f315203cda4 = "spectrum-Badge--small_a70f1a";
$700c1b49452b6d47$export$ce0abe6a13066aee = "spectrum-Badge--neutral_a70f1a";
$700c1b49452b6d47$export$250de4fc6219f4ba = "spectrum-Badge--negative_a70f1a";
$700c1b49452b6d47$export$f39824621eafe7d1 = "spectrum-Badge--yellow_a70f1a";
$700c1b49452b6d47$export$b9072ec5c3c3196e = "spectrum-Badge--seafoam_a70f1a";
$700c1b49452b6d47$export$9cb54bc972cacc06 = "spectrum-Badge--positive_a70f1a";
$700c1b49452b6d47$export$a276b7cecb1e7608 = "spectrum-Badge--info_a70f1a";
$700c1b49452b6d47$export$17d8c34537918c7 = "spectrum-Badge--fuchsia_a70f1a";
$700c1b49452b6d47$export$cf21c81baa11bb0f = "spectrum-Badge--indigo_a70f1a";
$700c1b49452b6d47$export$a6258e5c939933e = "spectrum-Badge--magenta_a70f1a";
$700c1b49452b6d47$export$67ff21277f017a3f = "spectrum-Badge--purple_a70f1a";




function $c2a9f60d7159b63f$var$Badge(props, ref) {
    let { children: children , variant: variant , ...otherProps } = (0, $ha4zE$useProviderProps)(props);
    let domRef = (0, $ha4zE$useDOMRef)(ref);
    let { styleProps: styleProps  } = (0, $ha4zE$useStyleProps)(otherProps);
    let isTextOnly = (0, $ha4zE$react).Children.toArray(props.children).every((c)=>!/*#__PURE__*/ (0, $ha4zE$react).isValidElement(c));
    return /*#__PURE__*/ (0, $ha4zE$react).createElement("span", {
        ...(0, $ha4zE$filterDOMProps)(otherProps),
        ...styleProps,
        role: "presentation",
        className: (0, $ha4zE$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($700c1b49452b6d47$exports))), "spectrum-Badge", {
            [`spectrum-Badge--${variant}`]: variant
        }, styleProps.className),
        ref: domRef
    }, /*#__PURE__*/ (0, $ha4zE$react).createElement((0, $ha4zE$SlotProvider), {
        slots: {
            icon: {
                size: "S",
                UNSAFE_className: (0, $ha4zE$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($700c1b49452b6d47$exports))), "spectrum-Badge-icon")
            },
            text: {
                UNSAFE_className: (0, $ha4zE$classNames)((0, (/*@__PURE__*/$parcel$interopDefault($700c1b49452b6d47$exports))), "spectrum-Badge-label")
            }
        }
    }, typeof children === "string" || isTextOnly ? /*#__PURE__*/ (0, $ha4zE$react).createElement((0, $ha4zE$Text), null, children) : children));
}
/**
 * Badges are used for showing a small amount of color-categorized metadata, ideal for getting a user's attention.
 */ let $c2a9f60d7159b63f$export$37acb3580601e69a = /*#__PURE__*/ (0, $ha4zE$forwardRef)($c2a9f60d7159b63f$var$Badge);




export {$c2a9f60d7159b63f$export$37acb3580601e69a as Badge};
